internal/stategen/cleanup.go

```diff
@@ -0,0 +1,225 @@
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(fileBytes), "\n")
+	var newLines []string
+	inPackages := false
+	inBlock := false
+	braceCount := 0
+	for _, line := range lines {
+		if strings.Contains(line, "\"packages\":") {
+			inPackages = true
+		}
+		if inPackages && strings.Contains(line, "\""+moduleName+"\":") {
+			inBlock = true
+		}
+
+		if inBlock {
+			braceCount += strings.Count(line, "{")
+			braceCount -= strings.Count(line, "}")
+			if braceCount == 0 {
+				inBlock = false
+			}
+			continue
+		}
+
+		// Handle manifest files and non-package blocks.
+		if strings.Contains(line, "\""+moduleName+"\":") {
+			// Check if it's the last line in a block without a trailing comma.
+			trimmedLine := strings.TrimSpace(line)
+			if !strings.HasSuffix(trimmedLine, ",") {
+				if len(newLines) > 0 {
+					lastLine := strings.TrimSpace(newLines[len(newLines)-1])
+					if strings.HasSuffix(lastLine, ",") {
+						newLines[len(newLines)-1] = strings.TrimSuffix(newLines[len(newLines)-1], ",")
+					}
+				}
+			}
+			continue
+		}
+
+		newLines = append(newLines, line)
+	}
+
+	output := strings.Join(newLines, "\n")
+	if bytes.Equal([]byte(output), fileBytes) {
+		return nil
+	}
+
+	return os.WriteFile(path, []byte(output), 0644)
+}
+
+// cleanupPostProcessorConfig removes the module and service-config entries for
+// a given module from the internal/postprocessor/config.yaml file.
+// NOTE: This function does not remove modules from manual-clients.
+func cleanupPostProcessorConfig(repoRoot, moduleName string) error {
+	configPath := filepath.Join(repoRoot, "internal", "postprocessor", "config.yaml")
+	fileBytes, err := os.ReadFile(configPath)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(fileBytes), "\n")
+	var newLines []string
+
+	// First pass: remove from modules list.
+	inModules := false
+	for _, line := range lines {
+		if strings.HasPrefix(line, "modules:") {
+			inModules = true
+		} else if !strings.HasPrefix(line, " ") { // top-level key
+			inModules = false
+		}
+		if inModules && strings.TrimSpace(line) == "- "+moduleName {
+			continue
+		}
+		newLines = append(newLines, line)
+	}
+
+	// Second pass: remove from service-configs.
+	lines = newLines
+	newLines = []string{}
+	inServiceConfigs := false
+	var serviceConfigBlock []string
+	isTargetServiceConfig := false
+	for _, line := range lines {
+		if strings.HasPrefix(line, "service-configs:") {
+			inServiceConfigs = true
+			newLines = append(newLines, line)
+			continue
+		}
+		if !strings.HasPrefix(line, " ") { // top-level key
+			if inServiceConfigs {
+				if !isTargetServiceConfig {
+					newLines = append(newLines, serviceConfigBlock...)
+				}
+				isTargetServiceConfig = false
+				serviceConfigBlock = nil
+			}
+			inServiceConfigs = false
+		}
+
+		if !inServiceConfigs {
+			newLines = append(newLines, line)
+			continue
+		}
+
+		// In service-configs section
+		if strings.HasPrefix(line, "  - input-directory:") {
```

Comment:
Are we certain that all entries start with `input-directory`? (The order shouldn't matter in general YAML)

Reply:
Yes, they do.